### PR TITLE
Docstring fixes related to Niimg-like objects

### LINKED ISSUE #383: Remaining inconsistent niimg-like docstrings
Still some are remaining after #294.

The compromise at the time was to use something like this consistently:

```
niimg: Niimg-like object
    See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
```

A hacky one-liner to find a few replacement candidates:

```
~/dev/nilearn $ git grep -n -i -P 'nifti.like|nifti[^1]image' nilearn | grep -v test
nilearn/_utils/niimg_conversions.py:80:        # Copy locally the nifti_image to avoid the side effect of data
nilearn/_utils/niimg_conversions.py:95:        Nifti image to copy.
nilearn/_utils/niimg_conversions.py:123:        If niimg is a string, consider it as a path to Nifti image and
nilearn/_utils/niimg_conversions.py:133:    result: nifti-like
nilearn/_utils/niimg_conversions.py:314:        If niimg is a string, consider it as a path to Nifti image and
nilearn/_utils/niimg_conversions.py:362:    # First image is supposed to be a path or a Nifti-like element
nilearn/_utils/niimg_conversions.py:372:        raise TypeError("Data must be either a 4D Nifti image or a"
nilearn/_utils/niimg_conversions.py:373:                        " list of 3D Nifti images. You provided a %s%dD"
nilearn/decomposition/multi_pca.py:161:    `mask_img_`: Nifti like image
nilearn/image/resampling.py:2:Utilities to resample a Nifti Image
nilearn/image/resampling.py:266:    """Resample a Nifti image
nilearn/image/resampling.py:272:        Path to a nifti file or nifti-like object
nilearn/image/resampling.py:483:            Path to a nifti file or nifti-like object
nilearn/input_data/base_masker.py:236:            The nifti-like images to fit and transform.
nilearn/input_data/multi_nifti_masker.py:92:    mask_img_: Nifti like image
nilearn/input_data/nifti_masker.py:90:    `mask_img_` : Nifti like image
nilearn/masking.py:31:    mask_img: nifti-like image
nilearn/masking.py:126:        grp_mask: 3D nifti-like image
nilearn/masking.py:349:    mask : 3D nifti-like image
nilearn/masking.py:500:    mask : 3D nifti-like image
nilearn/masking.py:528:    Read the time series from the given nifti images or filepaths,
nilearn/masking.py:684:    mask_img: nifti-like image
nilearn/masking.py:689:    data: nifti-like image (or list of)
nilearn/plotting/displays.py:372:                The nifti-image-like. If it is a masked array, only
nilearn/plotting/displays.py:418:                The Nifti-Image like object to plot
nilearn/plotting/img_plotting.py:2:Functions to do automatic visualization of nifti-like images
nilearn/plotting/img_plotting.py:122:            Path to a nifti file or nifti-like object
nilearn/plotting/img_plotting.py:297:        anat_img : a nifti-image like object or a filename
nilearn/plotting/img_plotting.py:373:        epi_img : a nifti-image like object or a filename
nilearn/plotting/img_plotting.py:443:        roi_img : a nifti-image like object or a filename
nilearn/plotting/img_plotting.py:446:        bg_img : a nifti-image like object or a filename
nilearn/plotting/img_plotting.py:517:        stat_map_img : a nifti-image like object or a filename
nilearn/plotting/img_plotting.py:519:        bg_img : a nifti-image like object or a filename
nilearn/plotting/img_plotting.py:625:        stat_map_img : a nifti-image like object or a filename
```

### PR DESCRIPTION
**Pull request by banilo**
attempts to fix #383

### CODE REVIEW

File: nilearn/_utils/niimg_conversions.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def check_niimg(niimg, ensure_3d=False):
     ----------
     niimg: Niimg-like object
         See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
-        If niimg is a string, consider it as a path to Nifti image and
+        If niimg is a string, consider it as a path to nifti image and
```
Comment (lesteve) [inline nilearn/_utils/niimg_conversions.py]: *a* nifti image or maybe Nifti image not sure what the convention about the capitalization of nifti is.

File: nilearn/_utils/niimg_conversions.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ def check_niimg(niimg, ensure_3d=False):
 
     Returns
     -------
-    result: nifti-like
+    result: Niimg-like object
        result can be nibabel.Nifti1Image or the input, as-is. It is guaranteed
```
Comment (lesteve) [inline nilearn/_utils/niimg_conversions.py]: missing URL

Comment (lesteve) [inline nilearn/_utils/niimg_conversions.py]: Oops forget that it is present a few lines below

File: nilearn/_utils/niimg_conversions.py [hunk 3]
```diff
@@ -369,8 +369,7 @@ def check_niimgs(niimgs, accept_3d=False, return_iterator=False):
     if (dim + depth) != 4:
         # Detailed error message that tells exactly the user what
         # was provided and what should have been provided.
-        raise TypeError("Data must be either a 4D Nifti image or a"
-                        " list of 3D Nifti images. You provided a %s%dD"
+        raise TypeError("Data must a be Niimg-like object. You provided a %s%dD"
```
Comment (lesteve) [inline nilearn/_utils/niimg_conversions.py]: I think you need to say 4D Niimg-like object (and probably add the URL so that the error is helpful)

File: nilearn/image/image.py [hunk 4]
```diff
@@ -402,8 +401,6 @@ def mean_img(imgs, target_affine=None, target_shape=None,
 
     imgs: Niimg-like object or iterable of Niimg-like objects
         See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
-        One or several niimage(s), either 3D or 4D (note that these
-        can be file names).
```
Comment (lesteve) [inline nilearn/image/image.py]: I think we still need to have a short description of the parameter

File: nilearn/image/image.py [hunk 5]
```diff
@@ -478,7 +475,6 @@ def swap_img_hemispheres(img):
     ----------
     img: Niimg-like object
         See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
-        One or several niimgs.
```
Comment (lesteve) [inline nilearn/image/image.py]: idem, needs a short description of the parameter

File: nilearn/input_data/base_masker.py [hunk 6]
```diff
@@ -232,8 +232,8 @@ def fit_transform(self, X, y=None, confounds=None, **fit_params):
 
         Parameters
         ----------
-        X : niimgs
-            The nifti-like images to fit and transform.
+        X : niimg: Niimg-like object
```
Comment (lesteve) [inline nilearn/input_data/base_masker.py]: the first niimg and colon should be removed

File: nilearn/image/image.py (hunk 4 above)
Comment (banilo) [inline nilearn/image/image.py]: Adressed - but did not fold.

File: nilearn/image/image.py (hunk 5 above)
Comment (banilo) [inline nilearn/image/image.py]: Adressed - but did not fold.

File: nilearn/image/image.py [hunk 7]
```diff
@@ -218,7 +218,6 @@ def smooth_img(imgs, fwhm):
     ==========
     imgs: Niimg-like object or iterable of Niimg-like objects
         See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
-        One or several niimage(s), either 3D or 4D.
```
Comment (lesteve) [inline nilearn/image/image.py]: Same comment here about a short description needed

File: nilearn/masking.py [hunk 8]
```diff
@@ -28,7 +28,8 @@ def _load_mask_img(mask_img, allow_empty=False):
 
     Parameters
     ----------
-    mask_img: nifti-like image
+    mask_img: niimg: Niimg-like object
```
Comment (lesteve) [inline nilearn/masking.py]: One 'niimg: ' too many

File: nilearn/masking.py [hunk 9]
```diff
@@ -123,7 +124,7 @@ def intersect_masks(mask_imgs, threshold=0.5, connected=True):
 
     Returns
     -------
-        grp_mask: 3D nifti-like image
+        grp_mask: nibabel.Nifti1Image
```
Comment (lesteve) [inline nilearn/masking.py]: Shouldn't you keep 3D ?

File: nilearn/masking.py [hunk 10]
```diff
@@ -346,7 +347,7 @@ def compute_multi_epi_mask(epi_imgs, lower_cutoff=0.2, upper_cutoff=0.85,
 
     Returns
     -------
-    mask : 3D nifti-like image
+    mask : nibabel.Nifti1Image
```
Comment (lesteve) [inline nilearn/masking.py]: Keep 3D?

File: nilearn/masking.py [hunk 11]
```diff
@@ -497,7 +498,7 @@ def compute_multi_background_mask(data_imgs, border_size=2, upper_cutoff=0.85,
...
```
Comment (lesteve) [inline nilearn/masking.py]: Keep 3D?

File: nilearn/image/image.py (hunk 7 above)
Comment (banilo) [inline nilearn/image/image.py]: Done

**Comment by lesteve** [comment]
LGTM, merging, thanks.

Next time better commit messages and it'll be even greater :wink:
